File: src/Longest Common Prefix/Longest Common Prefix.cpp
class Solution {
	public:
		string longestCommonPrefix(vector<string> &strs) {
			if(!strs.size()) return "";
			if(strs.size() == 1) return strs[0];

			int pos = 0, len = strs[0].length();
			while(pos < len)
			{
				for(int i = 1 ; i < strs.size() ; i++)
				{
					if(strs[i][pos] != strs[0][pos]) return strs[0].substr(0, pos); 
				}
				pos++;
			}
			return strs[0];
		}
};

File: src/Longest Common Prefix/Longest Common Prefix.py
class Solution:
	# @return a string
	def longestCommonPrefix(self, strs):
		if not strs: return ""
		if len(strs) == 1: return strs[0]
	
		s1, s2 = min(strs), max(strs)
		for index, c in enumerate(s1):
			if s2[index] != c: return s1[:index:]
		return s1
